src/Service/CommunicationService.php

```diff
@@ -3,18 +3,21 @@
 namespace App\Service;
 
 use Exception;
-use Makaira\HttpClient;
+use Doctrine\Persistence\ManagerRegistry;
+use App\Entity\AppInfo;
 
 class CommunicationService
 {
     private string $nonce;
+    private ManagerRegistry $doctrine;
 
     /**
      * @throws Exception
      */
-    public function __construct(private readonly string $clientSecret, private readonly HttpClient $httpClient, private readonly string $url, private readonly string $instance)
+    public function __construct(ManagerRegistry $doctrine)
```

Comment:
some as above, inject the Repository directly

Reply:
Updated